contracts/modules/STO/USDTieredSTO.sol

```diff
@@ -266,8 +266,10 @@ contract USDTieredSTO is USDTieredSTOStorage, STO {
                 tiers[i].mintedTotal = tiers[i].tokenTotal;
             }
         }
-        require(ISecurityToken(securityToken).mint(reserveWallet, tempReturned), "Minting Failed");
-        emit ReserveTokenMint(msg.sender, reserveWallet, tempReturned, currentTier);
+        address _wallet = (treasuryWallet == address(0) ? IDataStore(getDataStore()).getAddress(TREASURY) : treasuryWallet);
+        require(_wallet != address(0));
```

Comment:
Is there a case where an issuer would not want to automatically mint unsold tokens to a treasury wallet? In a STO implementation I am working on I have it so if the reserve wallet is set to address zero the STO will finalize but won't mint unsold tokens.
I've been trying to keep my STO up to date with changes in the dev 3.0.0 branch but this one would mean I'd require both the STO configured treasury wallet and global treasury wallet to be zero for minting unsold tokens to be disabled. Depending on what else the global treasury wallet is used for this might make disabling minting unsold tokens in this manner impossible. 
This only impacts my STO contract currently so I hope you don't mind my comment here. I guess the issuer always has the option to burn minted tokens they don't want but though I'd mention this point for you to consider. I may just add a function for finalizeMintUnsoldToTreasury and finalizeWitoutMinting instead to maintain this option while aligning with this change (once merged to Dev-3.0.0).

Reply:
@F-OBrien The Idea is very same as the earlier implementation has, Issuer has to mint the unsold token to reserve/local treasury wallet/global treasury wallet. Technically yes you could work with one more boolean variable. If we think of a more legal perspective as issuer committed to issuing `X` amount of supply in STO (on legal documents) then issuer has to mint that much amount of tokens by considering that issuer can burn those in another transaction by giving some legal statement. It will help to maintain the ledger of legal documents. I am not quite sure though but it looks cleaner.